src/System.Management.Automation/engine/parser/Compiler.cs

```diff
@@ -4973,6 +4981,20 @@ public Expression GenerateCallContains(Expression lhs, Expression rhs, bool igno
                 rhs.Cast(typeof(object)));
         }
 
+        public object VisitTernaryExpression(TernaryExpressionAst ternaryExpressionAst)
+        {
+            var expr = Expression.Condition(
+                CaptureAstResults(
```

Comment:
>CaptureAstResults [](start = 16, length = 17)

`CaptureAstResults` is not typically needed for most conditions and is fairly expensive.

Reply:
Were you thinking about `CaptureStatementResults`?
I used `CaptureAstResults` intentionally to avoid the unnecessary things in the `CaptureStatementResults`. What `CaptureAstResults` does in this case is just replacing current pipe with a temp pipe (list) and then convert the expression result to a bool value using `PSPipelineResultToBoolBinder`.

Reply:
@lzybkr Are you suggesting using `LanguagePrimitive.IsTrue(object obj)` here to replace `CaptureAstResults`? Like:
```
var expr = Expression.Condition(
    Expression.Call(
        CachedReflectionInfo.LanguagePrimitives_IsTrue,
        Compile(ternaryExpressionAst.Condition)),
    Compile(ternaryExpressionAst.IfTrue).Convert(typeof(object)),
    Compile(ternaryExpressionAst.IfFalse).Convert(typeof(object)));
```
From your experience, do you think this will be less expensive than using `CaptureAstResults`?

Reply:
If you look closely at the generated code for `CaptureAstResults`, it should be obvious why you'd want to avoid it (exception handling plus temporary lists), especially if most conditions are simple `bool` values.

I'm don't have a precise suggestion, but you should look closely at how conditions work elsewhere.  You'll see that they sometimes so call `CaptureAstResults`, but they avoid it if possible.

Reply:
You are right. 
I think `Convert(typeof(bool))` should just do it. It should be less expensive than calling `LanguagePrimitives.IsTrue` here, since the former uses the dynamic site `PSConvertBinder` that will cache the to-bool conversions while for the latter, it has to do the figuring-out part every time.